File: YACS Python Solution/2020.10 YACS 丙组乙组甲组/YACS 2020.10 丙组 #4.py
n = int(input())
a,b = list(map(int,input().split())),list(map(int,input().split()))

minprice = a[0]
maxprofit = 0
for i in range(1,n):
    minprice = min(minprice, a[i])
    maxprofit = max(maxprofit, b[i] - minprice)
print(maxprofit)

File: YACS Python Solution/2020.10 YACS 丙组乙组甲组/YACS 2020.10 丙组 #4.cpp
#include<bits/stdc++.h>
using namespace std;
int a[2000002],b[2000002],n,minprice,maxprofit;


int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);    
    cin>>n;
    for (int i = 0; i < n; i++) cin>>a[i];
    for (int i = 0; i < n; i++) cin>>b[i];

    minprice = a[0];
    maxprofit = 0;

    for (int i = 1; i < n; i++)
    {
        minprice = min(minprice, a[i]);
        maxprofit = max(maxprofit, b[i] - minprice);
    }
    cout<<maxprofit<<endl;
    return 0;
}